### ISSUE TITLE
Protect section with MFA

### ISSUE THREAD

**Issue by Pyrobolser** [aspnet/AspNetCore.Docs#15791]
How to force a user to connect with MFA to access some part of the application. The documentation
explicit quote this example:

>For example, two cookie handlers where one contains a basic identity and one is created when a multi-factor authentication (MFA) has been triggered. MFA may be triggered because the user requested an operation that requires extra security.

But doesn't really go any further about this. I have been trying to find more information about how to implement this properly but was unable to find any relevant documentation. If you could integrate some code sample on the subject or point me in the right direction.

Thank you,

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ee71441f-471a-89cd-0ef4-9b32b5dd85cf
* Version Independent ID: 0096c365-2f4e-5d21-89a0-c4f10fcb0ed9
* Content: [Authorize with a specific scheme in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authorization/limitingidentitybyscheme?view=aspnetcore-3.0)
* Content Source: [aspnetcore/security/authorization/limitingidentitybyscheme.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/security/authorization/limitingidentitybyscheme.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by blowdart** [comment]
We've never actually written any code for this. @HaoK it should be possible though right?

**Comment by Rick-Anderson** [comment]
@HaoK  also see Master issue for: Authorize with a specific scheme #16190

**Comment by Rick-Anderson** [comment]
@HaoK  can you outline the steps so I can update the article?

**Comment by HaoK** [comment]
It shouldn't be dramatically different than the `[Authorize("Admin")]` scenario that requires elevated privileges, the only complication is that it will likely require an additional scheme representing the MFA, so that its possible to challenge only for the MFA scheme (like prompting for an authenticator code, or sending a two factor code).  SignInManager does this underneath the hood in identity, see https://github.com/dotnet/aspnetcore/blob/master/src/Identity/Core/src/SignInManager.cs#L790

Basically after a password check has been done, it looks for the presence of a Tfa cookie, if it doesn't exist, it returns a Tfa required result, and the page is responsible for sending the user to the appropriate place.  If you aren't using identity, you just need to mimic that flow using Authenticate("YourMFAScheme") to determine if its needed.

Does that make sense?
